runtime/mount/kernel.py

```diff
@@ -1084,6 +1085,38 @@ async def send_globals_summary(self) -> None:
 
         await self.send({"type": "globals_summary", "summary": summary})
 
+    async def upload_ldata(
+        self,
+        dst: str,
+        key: str,
+        viewer_id: str | None,
+        cell_id: str | None,
+        filename: str | None,
+    ) -> tuple[bool, str | None]:
+        df = self.k_globals[key]
+
+        pagination_settings = self.lookup_pagination_settings(
+            cell_id=cell_id, viewer_id=viewer_id, source_key=key
+        )
+
+        if df is None or not isinstance(df, pd.DataFrame):
+            return False, "viewer does not exists or variable is not a pandas DataFrame"
+
+        res = filter_and_sort(df=df, pagination_settings=pagination_settings)
+
+        tmp_dir = Path.home() / ".latch" / "plots"
+        tmp_dir.mkdir(parents=True, exist_ok=True)
+
+        name = filename if filename is not None else key
+        local_path = tmp_dir / f"./{name}.csv"
+        res.to_csv(local_path, index=False)
+
+        if not local_path.exists():
+            return False, "unable to save dataframe to csv"
+
+        LPath(dst).upload_from(local_path)
```

Comment:
ideally we would stream-upload directly from the dataframe without a temp file
add a todo?